### PR TITLE
Update randomizer interface for V3 core

### PR DESCRIPTION
**Pull request by ryley-o**
## Only merge after #230 

Update randomizer interface for V3 core contract.

This frees some constraints on our randomizer contract at the cost of some additional gas during mints.

## Gas impacts
Mint costs are increased, as shown in the table below.

_(All gas costs assume project 3 mint 501 of 1000)_
| Gas Test | Previous gas used to purchase | New gas used to purchase | % change |
| --- | --- | --- | --- |
| MinterSetPriceV2_V3Core | 143687 | 146523 | +2.0% |
| MinterDAExpV2_V3Core | 155411 | 158247 | +1.8% |

## Tests

- [x] Update existing tests 
- [x] Add new tests for new functionality

### CODE REVIEW

**Comment by ryley-o** [comment]
@jakerockland - would appreciate your thoughts on this one

File: contracts/BasicRandomizerV2.sol [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    function assignCoreAndRenounce(address _genArt721Core) external onlyOwner {
+        renounceOwnership();
+        genArt721Core = IGenArt721CoreContractV3(_genArt721Core);
+    }
+
+    // When `genArt721Core` calls this, it can be assured that the randomizer
+    // will set a bytes32 hash for tokenId `_tokenId` on the core contract.
+    function assignTokenHash(uint256 _tokenId) external {
+        uint256 time = block.timestamp;
```
Comment (jakerockland) [inline contracts/BasicRandomizerV2.sol:23]: I think this should have an assertion that the `msg.sender` is `genArt721Core` ya?

Comment (jakerockland) [reply contracts/BasicRandomizerV2.sol:23]: Ah nevermind, I see we have the assertion in `setTokenHash_8PT`. That said, probably worth a dev documentation comment making this assumption by the randomizer and explicit one.

**Review by jakerockland — APPROVED** [review]

File: contracts/BasicRandomizerV2.sol (hunk 1 above)
Comment (ryley-o) [reply contracts/BasicRandomizerV2.sol]: no you are correct - this does need that assertion. I implemented the fix last night but forgot to push the commit! updated!

Comment (jakerockland) [reply contracts/BasicRandomizerV2.sol]: Ah yeah the assertion definitely does need to go both ways 😅 

I probably shouldn't review PRs before coffee. TY for updating!

**Comment by jakerockland** [comment]
Only follow up here is storing randomizer as a push-only type stack, vs. as a single field, per our sync today–though that can also be done in a separate PR and we can merge this one without that change 👍 😄

**Comment by ryley-o** [comment]
>Only follow up here is storing randomizer as a push-only type stack, vs. as a single field, per our sync today–though that can also be done in a separate PR and we can merge this one without that change 

see #235 👍

**Pull request merged**